web-console/knikubevirt/snapshots/snapshots.md

```diff
@@ -0,0 +1,75 @@
+# Virtual Machines Snapshots
+
+Snapshots allow users to create a copy of the VM disks and memory, mainly for restore and backup purposes. The snapshots are taken live or offline depending on if the VM is running or not. 
+This document describes the snapshot design flow.  
+
+
+
+
+## From VM List 
+
+![Offline VM list - snapshot action from a single VM kebab](img/1-0.jpg)
+When clicking on the kebab menu in the VM list, the user will be able to take a snapshot of the VM, meaning he will copy the disks, memory content, and metadata for restoring, backup, creating templates and cloning VMs.
+
+![Offline Snapshotting status indication](img/1-1.jpg)
+Any Virtual machine that has a snapshot in progress will show this in the state:
+When in this progress mode all actions would be disabled and the “cancel snapshot” would be available.
+
+
+## Snapshot modal
+
+![Offline Shanpshot modal - defalut](img/2-0.jpg)
+In the basic modal, the snapshot will take all current configuration and selected parameters.
+The user must add a descriptive name.
+
+![Offline Snapshot modal - advanced](img/2-1.jpg)
+![Offline Snapshot modal - dropdown](img/2-2.jpg)
+![Offline Snapshot modal - dropdown default](img/2-3.jpg)
+![Offline Snapshot modal - dropdown change](img/2-4.jpg)
+
+In the advanced menu, the user will be able to decide if to include all disks or just some of them and if to include memory as part of the snapshot or not.
+
+## Scheduled snapshots
```

Comment:
@beanh66 - Think we should call out future additions like this? Or should we just leave it out and add as it's ready?

Reply:
@matthewcarleton mentioned that we have this behavior on PF4, and there is a follow-up ticket [207](https://github.com/openshift/openshift-origin-design/issues/207) to update this design to PF4, so I think we should leave it here.